tools/imageregistry/setup_imageregistry.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+kubectl create ns rukpak-system
+
+# create self-signed certificate and authorization secrets for registry server
+
+mkdir -p /tmp/var/imageregistry/certs /tmp/var/imageregistry/auth
+openssl req -x509 -newkey rsa:4096 -days 365 -nodes -sha256 -keyout /tmp/var/imageregistry/certs/tls.key -out /tmp/var/imageregistry/certs/tls.crt -subj "/CN=docker-registry" -addext "subjectAltName = DNS:docker-registry"
+docker run --rm --entrypoint htpasswd registry:2.6.2 -Bbn myuser mypasswd > /tmp/var/imageregistry/auth/htpasswd
+
+kubectl create secret tls certs-secret --cert=/tmp/var/imageregistry/certs/tls.crt --key=/tmp/var/imageregistry/certs/tls.key -n rukpak-system
+kubectl create secret generic auth-secret --from-file=/tmp/var/imageregistry/auth/htpasswd -n rukpak-system
+kubectl create configmap trusted-ca -n kube-system --from-file=ca.crt=/tmp/var/imageregistry/certs/tls.crt
+
+# create image registry service
+kubectl apply -f tools/imageregistry/service.yaml -n rukpak-system
+
+# set local variables
+export REGISTRY_NAME="docker-registry"
+export REGISTRY_IP=$(kubectl get service docker-registry -n rukpak-system -o jsonpath='{ .spec.clusterIP }')
+export REGISTRY_PORT=5000
+export NODE_PORT=$(kubectl get service docker-registry -n rukpak-system  -o jsonpath='{ .spec.ports[0].nodePort }')
+export NODE_IP=$(kubectl get node kind-control-plane -n rukpak-system -o jsonpath='{ .status.addresses[?(@.type=="InternalIP")].address }')
+
+#### Warning ####
+# Update test host configuration
+
+echo "$NODE_IP $REGISTRY_NAME" >> /etc/hosts
+mkdir -p /etc/docker/certs.d/$REGISTRY_NAME:$NODE_PORT
+cp /tmp/var/imageregistry/certs/tls.crt /etc/docker/certs.d/$REGISTRY_NAME:$NODE_PORT/ca.crt
```

Comment:
It concerns me that this setup changes my workstation's docker configuration. I get that this is necessary to ensure the local docker client can connect to the in-cluster registry and I see that `cleanup.sh` is what we'd use to restore our dev workstations, but I wonder if there's a fully in-cluster way to do this, such that `kind delete cluster --name=rukpak-e2e` is enough to blow it all away?

No idea if this would work, but would it be possible to run the registry image in the cluster and have it mount the kind node's image storage directories such that anything we `kind load` is available via the registry?

Or perhaps something like running a `podman` pod in the cluster that mounts its containerd storage/config from the node and can then push kind-node images to the in-cluster registry?

Or something like this: https://github.com/tilt-dev/ctlptl#kind-with-a-built-in-registry-at-a-pre-determined-port (based on https://kind.sigs.k8s.io/docs/user/local-registry/)

Reply:
I understand your concern. I don't like to affect the host system configuration, either.  I'll look into if there is a way to load the image uploaded by `kind load` in the registry instead of pushing it from the docker client on the host.

Reply:
I did a little digging and was able to successfully get at the kind node's local images. Here's a pod spec that mounts the containerd directories from the kind node, and runs `nerdctl --namespace k8s.io images`.

I assume we could tweak this to do `nerdctl push` to the local registry running in cluster?

<details>
  <summary>nerdctl.yaml</summary>

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: nerdctl
spec:
  containers:
  - command:
    - nerdctl
    - --namespace=k8s.io
    - images
    image: ghcr.io/containerd/nerdctl
    imagePullPolicy: Always
    name: nerdctl
    volumeMounts:
    - mountPath: /run/containerd
      name: run-containerd
    - mountPath: /var/lib/containerd
      name: var-lib-containerd
  volumes:
  - name: run-containerd
    hostPath:
      path: /run/containerd
  - name: var-lib-containerd
    hostPath:
      path: /var/lib/containerd
```

</details>

Reply:
I was looking into it but I haven't found one. I'll look at it. Thanks!